src/components/GeneralInfo/SystemCard/SystemCard.test.js

```diff
@@ -508,4 +448,376 @@ describe('SystemCard', () => {
+            {
+              instance_name: 'NOVA-X, ENV-Δ42',
+              product: 'HyperIRIS',
+              version: '3021.∞, nebula.7',
+            },
+          ],
+        },
+      },
+      expectedText: 'InterSystems',
+    },
+    {
+      name: 'IBM Db2',
+      workloads: {
+        ibm_db2: {
+          is_running: true,
+        },
+      },
+      expectedText: 'IBM Db2',
+    },
+    {
+      name: 'IBM Db2',
+      workloads: {
+        ibm_db2: {
+          is_running: false,
+        },
+      },
+      expectedText: 'IBM Db2',
+    },
+    {
+      name: 'CrowdStrike',
+      workloads: {
+        crowdstrike: {
+          falcon_aid: 'xfoCshFVO6TwXdGHvy',
+          falcon_backend: 'dOqLegz0W8159Q0X2, fNbEf-pmK, r1zrECSYr_FgUDMbIu2',
+          falcon_version: 'lsSRGjjnhpl2Cz-, -KPJKI_kSyHVP5khj',
+        },
+      },
+      expectedText: 'CrowdStrike',
+    },
+    {
+      name: 'Microsoft SQL',
+      workloads: {
+        mssql: {
+          version: 'nTQNi8yZSTEN, x_OkwFDlxq7dl, LkIh__hO0qTnYZoCwL',
+        },
+      },
+      expectedText: 'Microsoft SQL',
+    },
+    {
+      name: 'Oracle DB',
+      workloads: {
+        oracle_db: {
+          is_running: true,
+        },
+      },
+      expectedText: 'Oracle Database',
+    },
+    {
+      name: 'Oracle DB',
+      workloads: {
+        oracle_db: {
+          is_running: false,
+        },
+      },
+      expectedText: 'Oracle Database',
+    },
+  ];
+
+  describe('SystemCard workload rendering', () => {
+    it.each(workloadTestCases)(
+      'should render correctly with $name workload',
+      ({ workloads, expectedText }) => {
+        render(
+          <TestWrapper
+            store={mockStore({
+              ...initialState,
+              systemProfileStore: {
+                systemProfile: {
+                  loaded: true,
+                  ...testProperties,
+                  workloads,
+                },
+              },
+            })}
+          >
+            <SystemCard />
+          </TestWrapper>
+        );
+
+        expect(
+          screen.getByRole('definition', {
+            name: /Workloads value/i,
+          })
+        ).toHaveTextContent(expectedText);
+      }
+    );
+
+    it('should render correctly with every workload', () => {
+      render(
+        <TestWrapper
+          store={mockStore({
+            ...initialState,
+            systemProfileStore: {
+              systemProfile: {
+                loaded: true,
+                ...testProperties,
+                workloads: mockWorkloadsData,
+              },
+            },
+          })}
+        >
+          <SystemCard />
+        </TestWrapper>
+      );
+
+      expect(
+        screen.getByRole('definition', {
+          name: /Workloads value/i,
+        })
+      ).toHaveTextContent(
+        /SAP|Ansible|CrowdStrike|RHEL AI|InterSystems|IBM Db2|Microsoft SQL|Oracle Database/
+      );
+    });
+  });
+
+  const workloadClickTestCases = [
```

Comment:
Perhaps, we could move it to some separate fixtures file?

Reply:
Did that :)